bgpd/bgp_clist.c

```diff
@@ -718,125 +718,137 @@ community_regexp_delete (struct community *com, regex_t * reg)
 
 	for (entry = list->head; entry; entry = entry->next) {
 		if (entry->any)
-			return entry->direct == COMMUNITY_PERMIT ? 1 : 0;
+			return entry->direct == COMMUNITY_PERMIT ? true : false;
 
 		if (entry->style == COMMUNITY_LIST_STANDARD) {
 			if (community_include(entry->u.com, COMMUNITY_INTERNET))
-				return entry->direct == COMMUNITY_PERMIT ? 1
-									 : 0;
+				return entry->direct == COMMUNITY_PERMIT
+					       ? true
+					       : false;
 
 			if (community_match(com, entry->u.com))
-				return entry->direct == COMMUNITY_PERMIT ? 1
-									 : 0;
+				return entry->direct == COMMUNITY_PERMIT
+					       ? true
+					       : false;
 		} else if (entry->style == COMMUNITY_LIST_EXPANDED) {
 			if (community_regexp_match(com, entry->reg))
-				return entry->direct == COMMUNITY_PERMIT ? 1
-									 : 0;
+				return entry->direct == COMMUNITY_PERMIT
+					       ? true
+					       : false;
 		}
 	}
-	return 0;
+	return false;
 }
 
-int lcommunity_list_match(struct lcommunity *lcom, struct community_list *list)
+bool lcommunity_list_match(struct lcommunity *lcom, struct community_list *list)
 {
 	struct community_entry *entry;
 
 	for (entry = list->head; entry; entry = entry->next) {
 		if (entry->any)
-			return entry->direct == COMMUNITY_PERMIT ? 1 : 0;
+			return entry->direct == COMMUNITY_PERMIT ? true : false;
 
 		if (entry->style == LARGE_COMMUNITY_LIST_STANDARD) {
 			if (lcommunity_match(lcom, entry->u.lcom))
-				return entry->direct == COMMUNITY_PERMIT ? 1
-									 : 0;
+				return entry->direct == COMMUNITY_PERMIT
+					       ? true
+					       : false;
 		} else if (entry->style == LARGE_COMMUNITY_LIST_EXPANDED) {
 			if (lcommunity_regexp_match(lcom, entry->reg))
-				return entry->direct == COMMUNITY_PERMIT ? 1
-									 : 0;
+				return entry->direct == COMMUNITY_PERMIT
+					       ? true
+					       : false;
 		}
 	}
-	return 0;
+	return false;
 }
 
 
 /* Perform exact matching.  In case of expanded large-community-list, do
  * same thing as lcommunity_list_match().
  */
-int lcommunity_list_exact_match(struct lcommunity *lcom,
-			       struct community_list *list)
+bool lcommunity_list_exact_match(struct lcommunity *lcom,
+				 struct community_list *list)
 {
 	struct community_entry *entry;
 
 	for (entry = list->head; entry; entry = entry->next) {
 		if (entry->any)
-			return entry->direct == COMMUNITY_PERMIT ? 1 : 0;
+			return entry->direct == COMMUNITY_PERMIT ? true : false;
 
 		if (entry->style == LARGE_COMMUNITY_LIST_STANDARD) {
 			if (lcommunity_cmp(lcom, entry->u.com))
-				return entry->direct == COMMUNITY_PERMIT ? 1
-									 : 0;
+				return entry->direct == COMMUNITY_PERMIT
+					       ? true
+					       : false;
 		} else if (entry->style == LARGE_COMMUNITY_LIST_EXPANDED) {
 			if (lcommunity_regexp_match(lcom, entry->reg))
-				return entry->direct == COMMUNITY_PERMIT ? 1
-									 : 0;
+				return entry->direct == COMMUNITY_PERMIT
+					       ? true
+					       : false;
```

Comment:
All of these should be changed to return the cmp itself, eg `return entry->direct == COMMUNITY_PERMIT;`

Reply:
will fix